src/window.py

```diff
@@ -138,7 +138,10 @@ def _setup(self):
         self.select_all.connect("clicked", self._on_select_all)
         self.select_none.connect("clicked", self._on_select_none)
         self.new_folder.connect("clicked", self._on_new_folder)
-        self.close_button.connect("clicked", self._on_button_closed)
+
+        self.handler_close_go_home = self.close_button.connect("clicked", self._on_button_closed_go_home)
+        self.close_button.handler_block(self.handler_close_go_home)
```

Comment:
These two lines you can move to the `failed` signal callback like...

```
self.close_button.connect("clicked", self._on_clicked_go_home)
self.close_button.handler_block(self._default_close_handler)
```

then...